src/main/java/com/epam/test_generator/entities/results/TestResult.java

```diff
@@ -109,9 +81,42 @@ public void setProject(Project project) {
 
     public void setSuits(List<SuitResult> suits) {
         this.suits = suits;
+        countTestResultStatistics(suits);
+        setStatus(calculateStatus(suits));
+        setDuration(calculateDuration(suits));
     }
 
     public Project getProject() {
         return project;
     }
+
+    /**
+     * Summarize amount of Passed, Skipped and Failed tests of Tests executions.
+     *
+     * @param results list of {@link AbstractResult}
+     */
+    private void countTestResultStatistics(List<? extends AbstractResult> results) {
+        int amountOfPassed = 0;
```

Comment:
That may be written even shorter in stream api paradigm

        Map<Status, Integer> amounts = results.stream()
            .collect(Collectors.groupingBy(AbstractResult::getStatus, 
                Collectors.reducing(0, e -> 1, Integer::sum)));
        setAmountOfPassed(amounts.getOrDefault(Status.PASSED, 0));
        setAmountOfFailed(amounts.getOrDefault(Status.FAILED, 0));
        setAmountOfSkipped(amounts.getOrDefault(Status.SKIPPED, 0));